### PR TITLE
refactor: add an e2e package to consolidate all e2e test from other packages.

### PR DESCRIPTION
**Pull request by AngelCastilloB**
# Context

We need to consolidate existing e2e test segregated in different packages (wallet, web-extension, cardano-services and blockfrost) under a new e2e package to share the config and to avoid circular dependencies.

# Proposed Solution

Refactor the e2e test and move them to a new separate package.

# Important Changes Introduced

e2e tests were moved to a new package.

_This PR is clone of  #305  with a cleaner commit history._

### CODE REVIEW

File: packages/web-extension/package.json [hunk 1]
```diff
@@ -1,95 +1,89 @@
...
-    "uuid": "^8.3.2",
-    "webextension-polyfill": "^0.8.0"
-  },
-  "files": [
-    "dist/*",
-    "!dist/tsconfig.tsbuildinfo",
-    "LICENSE",
-    "NOTICE"
-  ]
+    "name": "@cardano-sdk/web-extension",
```
Comment (rhyslbw) [inline packages/web-extension/package.json:2]: We need to work out why this is a complete diff. Maybe different line endings? Also, the indents are being modified

File: packages/e2e/package.json [hunk 2]
```diff
@@ -0,0 +1,99 @@
+{
```
Comment (rhyslbw) [inline packages/e2e/package.json:1]: The indent spacing here isn't correct. We're not linting json, so it has to be manually controlled at this point.

File: packages/core/package.json [hunk 3]
```diff
@@ -1,53 +1,53 @@
...
-    "@emurgo/cardano-serialization-lib-nodejs": "9.1.0",
-    "bech32": "^2.0.0"
-  },
-  "files": [
-    "dist/*",
-    "!dist/tsconfig.tsbuildinfo",
-    "LICENSE",
-    "NOTICE"
-  ]
+    "name": "@cardano-sdk/core",
```
Comment (rhyslbw) [inline packages/core/package.json:2]: Please revert the indent change

**Review by rhyslbw — CHANGES_REQUESTED** [review]

File: packages/web-extension/package.json (hunk 1 above)
Comment (AngelCastilloB) [reply packages/web-extension/package.json:2]: yes I think is the indent, I think when I was using vscode it was auto formatting the files, I disabled it, but maybe this file was already re indented completely

File: packages/core/package.json (hunk 3 above)
Comment (AngelCastilloB) [reply packages/core/package.json]: All identation issues wtih package.json should now be fixed

File: packages/e2e/docker-compose.yml [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+
+services:
+  local-testnet:
+    build:
+      context: ./local-network
+    ports:
+      - 3001:3001
+    volumes:
+      - ${PWD}/local-network/sockets:/root/sockets
+      - ./local-network/config/network/${NETWORK:-mainnet}:/config/nodes # Mount the config so we can update the genesis files and hashes.
```
Comment (rhyslbw) [inline packages/e2e/docker-compose.yml:11]:
```suggestion
      - ./local-network/config:/config/nodes # Mount the config so we can update the genesis files and hashes.
```
Is there a reason to support multiple network configurations?

File: packages/e2e/docker-compose.yml [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+      timeout: 5s
+      retries: 5
+    logging:
+      driver: "json-file"
+      options:
+        max-size: "200k"
+        max-file: "10"
+
+  cardano-node-ogmios:
+    image: cardanosolutions/cardano-node-ogmios:${CARDANO_NODE_OGMIOS_VERSION:-v5.1.0}-${NETWORK:-mainnet}
```
Comment (rhyslbw) [inline packages/e2e/docker-compose.yml:47]:
```suggestion
    image: cardanosolutions/cardano-node-ogmios:${CARDANO_NODE_OGMIOS_VERSION:-v5.1.0}
```

File: packages/e2e/docker-compose.yml [hunk 6]
```diff
@@ -0,0 +1,127 @@
...
+  cardano-wallet:
+    image: inputoutput/cardano-wallet:2022.5.27
+    volumes:
+      - wallet-${NETWORK}-db:/wallet-db
+      - node-ipc:/ipc
+      - ./local-network/config:/config
+    ports:
+      - 8090:8090
+    entrypoint: []                                                                            
+    command: bash -c "sleep 10 && ([[ $$NETWORK == \"mainnet\" ]] && $$CMD --mainnet) || ($$CMD --testnet /config/network/${NETWORK}/genesis/byron.json)"
```
Comment (rhyslbw) [inline packages/e2e/docker-compose.yml:102]: Can remove support for _mainnet_

File: packages/e2e/docker-compose.yml [hunk 7]
```diff
@@ -0,0 +1,127 @@
...
+  postgres_password:
+    file: ./local-network/placeholder-secrets/postgres_password
+  postgres_user:
+    file: ./local-network/placeholder-secrets/postgres_user
+volumes:
+  db-sync-data:
+  node-db:
+  node-ipc:
+  postgres-data:
+  wallet-mainnet-db:
```
Comment (rhyslbw) [inline packages/e2e/docker-compose.yml:126]:
```suggestion
  wallet-db:
```

File: packages/e2e/docker-compose.yml [hunk 8]
```diff
@@ -0,0 +1,127 @@
...
+  wallet-testnet-db:
```
Comment (rhyslbw) [inline packages/e2e/docker-compose.yml:127]:
```suggestion
```

**Review by rhyslbw — CHANGES_REQUESTED** [review]

File: packages/e2e/docker-compose.yml (hunk 4 above)
Comment (AngelCastilloB) [reply packages/e2e/docker-compose.yml:11]: Not needed since we are using our custom config

**Review by rhyslbw — APPROVED** [review]
Amazing work @AngelCastilloB :rocket:

**Pull request merged**